### DISCUSSION TITLE
Need some tips for building tab context menu for my extension..

### DISCUSSION THREAD

**Discussion by zeroarst** [piroor/treestyletab#3389] [General]
Hi.. @piroor 
I would like to ask you for some tips for my extension development...
I am building na extension that can search tabs and show results. The result is like a list and each item is like a tab. 
I want to be able to right click on the tab item, and show a context menu like right click on a tab. I cannot make it work. I know that I can create a menu with tab contextType, but how do I turn the element into a tab component so that Firefox treats it like a tab...

I found treestyletab extension is able to right click an element and show tab context menu. I took look the source code, still cannot figure out...
Would you mind giving me some tips/directions? Thank you.

**Comment by irvinm** [comment]
Have you seen this?  https://github.com/piroor/treestyletab/wiki/Fake-Context-Menu-API

I would think you would just populate the menu items of the context menu to be displayed and when someone clicks on the menu item, you would know what tab to create and you would simply issue a [tabs.create()](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/tabs/create) with the information you are tracking.

**Comment by piroor** [comment]
In short: you need to call [`menus.overrideContext()`](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/menus/overrideContext) with a `contextmenu` event listener.
Here are some blog articles about how to implement tab context menu on your custom UI:

* https://piro.sakura.ne.jp/latest/blosxom/mozilla/xul/2018-10-14_override-context-on-fx64.htm
* https://piro.sakura.ne.jp/latest/blosxom/mozilla/xul/2018-10-26_override-context-for-each-case.htm
